eng/Version.Details.xml

```diff
@@ -52,18 +52,18 @@
       <Sha>1999c8c8ab7473a7e1c5b7bdf5ba6d9a985a69cc</Sha>
       <SourceBuild RepoName="emsdk" ManagedOnly="true" />
     </Dependency>
-    <Dependency Name="Microsoft.Build" Version="17.8.0-preview-23471-08">
+    <Dependency Name="Microsoft.Build" Version="17.8.0">
```

Comment:
Why is this 17.8.0 instead of 17.8.0-preview-23472-04?

/cc @MichaelSimons

Reply:
`ReadSourceBuildIntermediateNupkgDependencies` is getting confused

```
Microsoft.SourceBuild.Intermediate.msbuild
    Sha = 6cdef424154c976f04802b101e6be6292f8a8897
    Version = [17.8.0]
    SourceBuildRepoName = msbuild
    Name = Microsoft.Build
    SourceBuildManagedOnly = true
    Uri = https://github.com/dotnet/msbuild
    ExactVersion = 17.8.0
```

Reply:
OK I see why. The only ref that has a SourceBuild element is the stable-branded one.

https://github.com/dotnet/sdk/blob/4c7f3a3be755801018eca12ee5a28f3f9d6c8002/eng/Version.Details.xml#L55-L67

. . . should that be moved to the sourcebuild intermediate package instead?

Reply:
> Why is this 17.8.0 instead of 17.8.0-preview-23472-04?

Oh wait, you're asking why we have a full stable version? Because we are stabilizing.